internals/plan/plan.go

```diff
@@ -317,6 +318,90 @@ const (
 	ActionIgnore   ServiceAction = "ignore"
 )
 
+// LogTargets specifies which log targets should receive a service's logs.
+// It is either an explicit list of targets, or one of the below keywords.
+type LogTargets struct {
+	Targets []string
+	Keyword string
+}
+
+const (
+	// Valid keywords for LogTargets
+	DefaultLogTargets = "default"
+	AllLogTargets     = "all"
+	NoLogTargets      = "none"
+)
+
+func (t *LogTargets) UnmarshalYAML(value *yaml.Node) error {
+	if value.Kind == yaml.ScalarNode {
+		// decode keyword
+		switch value.Value {
+		case DefaultLogTargets, AllLogTargets, NoLogTargets:
+			t.Keyword = value.Value
+			return nil
+		default:
+			return fmt.Errorf("invalid value %q for LogTargets", value.Value)
+		}
+	}
+
+	// unmarshal to []string
+	return value.Decode(&t.Targets)
+}
+
+// Copy returns a deep copy of this LogTargets struct.
+func (t *LogTargets) Copy() *LogTargets {
+	return &LogTargets{
+		Targets: append([]string(nil), t.Targets...),
+		Keyword: t.Keyword,
+	}
+}
+
+// Merge merges the fields from other into t.
+func (t *LogTargets) Merge(other *LogTargets) {
+	if other.Keyword != "" {
+		// anything <- keyword yields keyword
+		t.Keyword = other.Keyword
+		t.Targets = nil
+
+	} else {
+		if t.Keyword != "" {
+			// keyword <- [list] yields [list]
+			t.Keyword = ""
```

Comment:
With this logic `t.Keyword == "all"` is no longer `all`.

`Targets` and `Keyword` are not mutually exclusive.

The merge logic here should be:
```
if other.Keyword is less inclusive than t.Keyword {
    other.Keyword = t.Keyword
}
other.Targets = other.Targets + t.Targets
```

`""` < `"None"` < `"Default"` < `"All"`

Reply:
Not really sure what you mean here. `Targets` and `Keyword` *are* mutually exclusive, in that only one should be defined at a time.

Maybe you're debating the current behaviour that merging `all <- [list]` yields the `[list]`? Admittedly this is a bit quirky and I'm willing to change this if we think it's confusing / counterintuitive.